cl/api/utils.py

```diff
@@ -604,24 +606,48 @@ def check_webhook_failure_count_and_notify(
         1: True,  # Send first webhook failing notification
         2: False,
         3: False,
-        4: True,  # Send second webhook failing notification
-        5: False,
-        6: True,  # Send third webhook failing notification
+        4: False,
+        5: True,  # Send second webhook failing notification
+        6: False,
         7: True,  # Send webhook disabled notification
     }
-
-    current_try_counter = webhook_event.retry_counter
-    disabled = False
     webhook = webhook_event.webhook
-    if current_try_counter >= WEBHOOK_MAX_RETRY_COUNTER and webhook.enabled:
-        webhook.enabled = False
-        # Don't send notification email via signal in cl.users.signals
-        webhook.save(update_fields=["enabled"])
-        disabled = True
+    if not webhook.enabled or webhook_event.debug:
+        return
 
+    webhook.failure_count = F("failure_count") + 1
+    update_fields = ["failure_count"]
+
+    current_try_counter = webhook_event.retry_counter
     notify = notify_on[current_try_counter]
     if notify:
-        notify_failing_webhook.delay(webhook_event.webhook.pk, disabled)
+        disabled = False
+        all_enqueued_for_retry_we = WebhookEvent.objects.filter(
+            webhook=webhook_event.webhook,
+            event_status=WEBHOOK_EVENT_STATUS.ENQUEUED_RETRY,
+            debug=False,
+        ).order_by("date_created")
+        all_enqueued_for_retry_we_list = list(all_enqueued_for_retry_we)
```

Comment:
I think I'd skip this query completely. You use this query in two places below. 

First, like this:

```
all_enqueued_for_retry_we.update(
                event_status=WEBHOOK_EVENT_STATUS.ENDPOINT_DISABLED
            )
```

Just replace that with a filter-update chain. Something like:

```
WebhookEvent.objects.filter(...).update(...)
```

Second, you use `all_enqueued_for_retry_we` as:

```
oldest_enqueued_for_retry = all_enqueued_for_retry_we_list[0]
```
I'd just do a query for that instead, using `earliest('date_created')`. That'll take this code from:

1 big SQL SELECT
1 big SQL UPDATE

To:

1 bit SQL UPDATE
1 small SQL SELECT

It might be faster, I don't know, but it should be clearer too.

Reply:
Cool, this is better, applied the changes, thank you!